unified-runtime/scripts/core/PROG.rst

```diff
@@ -10,6 +10,26 @@
  Programming Guide
 ========================
 
+Adapters
+========
+
+The top level abstraction in the unified runtime is the adapter handle. It owns
+the platform handles as well as any backend specific global state, for instance
+if a backend has global setup/teardown type entry-points these will typically
+be tied to the adapter handle's constructor and destructor.
+
+Adapter handles are reference counted via ${x}AdapterRetain and
+${x}AdapterRelease. Once their reference count reaches zero any objects
+associated with the adapter in question should be considered invalid. Calling
+${x}AdapterGet after any adapter handle's reference count has reached zero will
```

Comment:
The [documentation](https://oneapi-src.github.io/unified-runtime/core/api.html#uradapterrelease) for urAdapterRelease disagrees with this being undefined behavior:

> When the reference count of the adapter reaches zero, the adapter may perform adapter-specififc resource teardown. Resources must be left in a state where it safe for the adapter to be subsequently reinitialized with [urAdapterGet](https://oneapi-src.github.io/unified-runtime/core/api.html#ur__api_8h_1aef22402ea83514d9598ccbe3ae769e8b).

I don't think there's a specific use case for that scenario though, so it should be safe to change if needed

Reply:
oh yeah I'll update that, I know for sure that implementations as they currently exist will break if you try to do that so best to reflect the reality of the situation

Reply:
updated